extension/parquet/parquet_reader.cpp

```diff
@@ -500,6 +512,40 @@ unique_ptr<BaseStatistics> ParquetColumnSchema::Stats(ParquetReader &reader, idx
 	return ParquetStatisticsUtils::TransformColumnStatistics(*this, columns, reader.parquet_options.can_have_nan);
 }
 
+static bool IsVariantType(const SchemaElement &root, const vector<ParquetColumnSchema> &children) {
```

Comment:
The spec says that the "group" (struct) should be annotated with the `VARIANT` `logicalType`
The files produced by apache/iceberg don't do this, from what I gathered they are identified by their structure.

To be able to identify the Variant group we copy this structural identification, but this should ideally be changed to actually follow the spec.

Reply:
Ideally we use the logical type here - maybe we can have a `legacy_variant_encoding` to support this for now using this method?

Maybe also check if whoever can read this suffers from the same issue w.r.t. detecting regular structs as this?